x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/integration/DataStreamLifecycleDownsamplingSecurityIT.java

```diff
@@ -0,0 +1,494 @@
+        SystemDataStreamWithDownsamplingConfigurationPlugin pluginWithIllegalSystemDataStream =
+            new SystemDataStreamWithDownsamplingConfigurationPlugin();
+        IllegalArgumentException illegalArgumentException = expectThrows(
+            IllegalArgumentException.class,
+            () -> pluginWithIllegalSystemDataStream.getSystemDataStreamDescriptors()
+        );
+        assertThat(
+            illegalArgumentException.getMessage(),
+            is("System data streams do not support downsampling as part of their lifecycle configuration")
+        );
+    }
+
+    private void waitAndAssertDownsamplingCompleted(String dataStreamName) throws Exception {
+        List<Index> backingIndices = getDataStreamBackingIndices(dataStreamName);
+        String firstGenerationBackingIndex = backingIndices.get(0).getName();
+        String oneSecondDownsampleIndex = "downsample-1s-" + firstGenerationBackingIndex;
+        String tenSecondsDownsampleIndex = "downsample-10s-" + firstGenerationBackingIndex;
+
+        Set<String> witnessedDownsamplingIndices = new HashSet<>();
+        clusterService().addListener(event -> {
+            if (event.indicesCreated().contains(oneSecondDownsampleIndex)
+                || event.indicesDeleted().stream().anyMatch(index -> index.getName().equals(oneSecondDownsampleIndex))) {
+                witnessedDownsamplingIndices.add(oneSecondDownsampleIndex);
+            }
+            if (event.indicesCreated().contains(tenSecondsDownsampleIndex)) {
+                witnessedDownsamplingIndices.add(tenSecondsDownsampleIndex);
+            }
+        });
+
+        client().execute(RolloverAction.INSTANCE, new RolloverRequest(dataStreamName, null)).actionGet();
+
+        assertBusy(() -> {
+            assertNoAuthzErrors();
+            // first downsampling round
+            assertThat(witnessedDownsamplingIndices.contains(oneSecondDownsampleIndex), is(true));
+        }, 30, TimeUnit.SECONDS);
+
+        assertBusy(() -> {
+            assertNoAuthzErrors();
+            assertThat(witnessedDownsamplingIndices.size(), is(2));
+            assertThat(witnessedDownsamplingIndices.contains(oneSecondDownsampleIndex), is(true));
+
+            assertThat(witnessedDownsamplingIndices.contains(tenSecondsDownsampleIndex), is(true));
+        }, 30, TimeUnit.SECONDS);
+
+        assertBusy(() -> {
+            assertNoAuthzErrors();
+            List<Index> dsBackingIndices = getDataStreamBackingIndices(dataStreamName);
+
+            assertThat(dsBackingIndices.size(), is(2));
+            String writeIndex = dsBackingIndices.get(1).getName();
+            assertThat(writeIndex, backingIndexEqualTo(dataStreamName, 2));
+            // the last downsampling round must remain in the data stream
+            assertThat(dsBackingIndices.get(0).getName(), is(tenSecondsDownsampleIndex));
+            assertThat(indexExists(oneSecondDownsampleIndex), is(false));
```

Comment:
Should we also assert `assertThat(indexExists(firstGenerationBackingIndex), is(false));` here? I think that's guaranteed at this point